addons/account/models/account_move.py

```diff
@@ -5079,6 +5079,16 @@ def is_internal_partner(partner):
         if msg_dict.get('subject') and msg_dict.get('body'):
             msg_dict['body'] = Markup('<div><div><h3>%s</h3></div>%s</div>') % (msg_dict['subject'], msg_dict['body'])
 
+        # Don't create the move if no attachment.
+        if len(msg_dict['attachments']) < 1:
+            mail_template = self.env.ref('account.email_template_mail_gateway_failed')
+            lang = (partners and partners[0].lang) or self.env.user.lang
```

Comment:
Hm, I don't think odoobot `lang` is relevant here, would say just don't set context if there's no viable lang to set and let the template use whatever it sees fit.